mlx/traceability.py

```diff
@@ -144,28 +144,31 @@ def run(self):
         if len(self.arguments) > 1:
             caption = self.arguments[1].replace('\n', ' ')
 
+        # Store item info
+        item = TraceableItem(targetid)
+        item.set_document(env.docname, self.lineno)
+        item.bind_node(targetnode)
+        item.set_caption(caption)
+        item.set_content('\n'.join(self.content))
         try:
-            # Store item info
-            item = TraceableItem(targetid)
-            item.set_document(env.docname, self.lineno)
-            item.bind_node(targetnode)
-            item.set_caption(caption)
-            item.set_content('\n'.join(self.content))
             env.traceability_collection.add_item(item)
+        except TraceabilityException as err:
+            report_warning(env, err, env.docname, self.lineno)
 
-            # Add found relationships to item. All relationship data is a string of
-            # item ids separated by space. It is splitted in a list of item ids
-            for rel in env.traceability_collection.iter_relations():
-                if rel in self.options:
-                    related_ids = self.options[rel].split()
-                    for related_id in related_ids:
+        # Add found relationships to item. All relationship data is a string of
+        # item ids separated by space. It is splitted in a list of item ids
+        for rel in env.traceability_collection.iter_relations():
+            if rel in self.options:
+                related_ids = self.options[rel].split()
+                for related_id in related_ids:
+                    try:
                         env.traceability_collection.add_relation(targetid, rel, related_id)
+                    except TraceabilityException as err:
```

Comment:
this does the trick: don't bail out (exception) if one of the target items does not exist, but continue on to the next one

Reply:
Looks Ok to me.